docs/src/index.md

```diff
@@ -219,7 +221,8 @@ x1 = input(nothing)
 x2 = map(x->println("x2"), x1)
 x3 = map(x->println("x3"), x1)
 x4 = map((x,y)->println("x4"), x2, x3)
-push!(Source(x1), nothing) # prints x2 x3 x4
+g, s1 = compile(x1)
+push!(g, s1, nothing) # prints x2 x3 x4
```

Comment:
This `push!` notation seems a little awkward.  If you. think of the graph as a Dict of Source => Value pairs then maybe a nice API might be setindex! instead?

```Julia
g[s1] = nothing
```